src/main/java/seedu/toluist/model/Task.java

```diff
@@ -272,16 +285,134 @@ public void setTaskPriority(TaskPriority priority) {
-        } else {
+            break;
+        default:
             throw new IllegalArgumentException("Task priority must be either 'low' or 'high'.");
         }
     }
 
     public LocalDateTime getCompletionDateTime() {
         return completionDateTime;
     }
+
+    //@@author A0127545A
+    public LocalDateTime getRecurringEndDateTime() {
+        return recurringEndDateTime;
+    }
+
+    public RecurringFrequency getRecurringFrequency() {
+        return recurringFrequency;
+    }
+
+    public RecurringFrequency toRecurringFrequency(String recurringFrequencyString) {
+        try {
+            return RecurringFrequency.valueOf(recurringFrequencyString.toUpperCase());
+        } catch (IllegalArgumentException | NullPointerException exception) {
+            throw new IllegalArgumentException("Invalid recurring frequency string");
+        }
+    }
+
+    public void setRecurringFrequency(String recurringFrequencyString) {
+        setRecurringFrequency(toRecurringFrequency(recurringFrequencyString));
+    }
+
+    public void setRecurringFrequency(RecurringFrequency recurringFrequency) {
+        if (isRecurring()) {
+            this.recurringFrequency = recurringFrequency;
+        } else {
+            setRecurring(recurringFrequency);
+        }
+    }
+
+    public void setRecurringEndDateTime(LocalDateTime recurringEndDateTime) {
+        if (isRecurring()) {
+            this.recurringEndDateTime = recurringEndDateTime;
+        } else {
+            throw new IllegalArgumentException("Non-recurring task cannot have recurring until end date.");
+        }
+    }
+
+    public void setRecurring(String recurringFrequencyString) {
+        setRecurring(toRecurringFrequency(recurringFrequencyString));
+    }
+
+    public void setRecurring(RecurringFrequency recurringFrequency) {
+        setRecurring(null, recurringFrequency);
+    }
+
+    public void setRecurring(LocalDateTime recurringEndDateTime, String recurringFrequencyString) {
+        setRecurring(recurringEndDateTime, toRecurringFrequency(recurringFrequencyString));
+    }
+
+    public void setRecurring(LocalDateTime recurringEndDateTime, RecurringFrequency recurringFrequency) {
+        if (recurringFrequency == null) {
+            throw new IllegalArgumentException("Recurring task must have a"
+                    + " frequency of 'daily', 'weekly', 'monthly' or 'yearly'.");
+        }
+        this.recurringEndDateTime = recurringEndDateTime;
+        this.recurringFrequency = recurringFrequency;
+    }
+
+    public void unsetRecurring() {
+        this.recurringEndDateTime = null;
+        this.recurringFrequency = null;
+    }
+
+    public boolean canUpdateToNextRecurringTask() {
+        if (!isRecurring()) {
+            return false;
+        }
+        if (recurringEndDateTime == null) {
+            return true;
+        }
+        if (endDateTime == null) {
+            return LocalDateTime.now().isBefore(recurringEndDateTime);
```

Comment:
hint hint: you have to check that the next occurence's date is before the repeat until date